### PR TITLE
Change Regex of SupportedParallelClusterVersions to support alpha and beta versions

### PR DESCRIPTION
**Pull request by chenwany**
### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -1236,7 +1236,9 @@ class SchedulerPluginRequirementsSchema(BaseSchema):
     supports_cluster_update = fields.Bool(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
     supported_parallel_cluster_versions = fields.Str(
         metadata={"update_policy": UpdatePolicy.UNSUPPORTED},
-        validate=validate.Regexp(r"^((>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+,\s*)*(>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+$"),
+        validate=validate.Regexp(
+            r"^((>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+[a-z]?[0-9]?,\s*)*(>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+[a-z]?[0-9]?$"
```
Comment (demartinofra) [inline cli/src/pcluster/schemas/cluster_schema.py]: probably it's even more correct to have `([a-z][0-9]+)*`

Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Changed. With the change we also support version like 3.1.0a111b12

Comment (demartinofra) [reply cli/src/pcluster/schemas/cluster_schema.py]: ah right it should be `?` not `*`

`^((>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+([a-z][0-9]+)?,\s*)*(>|<|>=|<=)?[0-9]+\.[0-9]+\.[0-9]+([a-z][0-9]+)?$`

Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Changed

**Review by demartinofra — APPROVED** [review]

**Pull request merged**